services/QuillLMS/client/app/bundles/PremiumHub/components/usage_snapshots/reportSubscriptionModal.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react'
+
+import { DropdownInput } from '../../../Shared/index';
+
+const MONTHLY = 'Monthly';
+const WEEKLY = 'Weekly';
+const ON = 'On';
+const OFF = 'Off';
+
+const ReportSubscriptionModal = ({ cancel, existingPdfSubscription, isOpen, save }) => {
+  if (!isOpen) return null;
+
+  const [isSubscribed, setIsSubscribed] = React.useState(existingPdfSubscription ? true : false);
+  const [frequency, setFrequency] = React.useState(existingPdfSubscription?.frequency || MONTHLY);
+
+  const frequencyOptions = [{ "value": MONTHLY, "label": MONTHLY }, { "value": WEEKLY, "label": WEEKLY }]
+
+  const handleFrequencyOptionChange = (e: { value: string }) => {
+    setFrequency(e.value)
+  }
+
+  const handleIsSubscribedOptionChange = (option: { target: { value: string } }) => {
+    setIsSubscribed(option.target.value === ON)
+  }
+
+  function handleSaveClick() { save(isSubscribed, frequency, existingPdfSubscription) }
+
+  function renderFrequencyOptions() {
+    const selectedFrequency = frequencyOptions.find(option => frequency === option.value)
+
+    return (
+      <div className="frequency-options">
+        <label className="frequency-label" htmlFor="frequency">
+          <span>Frequency</span>
+        </label>
+        <DropdownInput
+          className={`${isSubscribed ? '' : 'disabled'}`}
+          disabled={!isSubscribed}
+          handleChange={handleFrequencyOptionChange}
+          isSearchable={false}
+          label=""
+          options={frequencyOptions}
+          value={selectedFrequency}
+        />
+      </div>
+    )
+  }
+
+  function renderIsSubscribedAndFrequency() {
+    return (
+      <div className="is-subscribed-and-frequency">
+        {renderIsSubscribedOptions()}
+        {renderFrequencyOptions()}
+      </div>
+    )
+  }
```

Comment:
I think this is an interesting preference as I tend to just order alphabetically so I don't need to think about where things go.  However, I can understand your preference for ordering.  I'll move it per your request here but I think this would be an interesting item for discussion in engineering.